batcher/sweep_batcher.go

```diff
@@ -0,0 +1,562 @@
+		returnChan:       br.SweepReqs,
+		completeChan:     br.completeChan,
+		wallet:           br.wallet,
+		chainNotifier:    br.chainNotifier,
+		signerClient:     br.signerClient,
+		musig2SignSweep:  br.musig2ServerSign,
+		verifySchnorrSig: br.VerifySchnorrSig,
+		store:            br.store,
+	}
+
+	batch, err := NewBatch(cfg, batchKit)
+	if err != nil {
+		return nil, err
+	}
+
+	id, err := batch.insertAndAcquireID(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// We add the batch to our map of batches and start it.
+	br.batches[id] = batch
+
+	br.wg.Add(1)
+	go func() {
+		defer br.wg.Done()
+
+		err := batch.Run(ctx)
+		if err != nil {
+			select {
+			case br.errChan <- err:
+				return
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
+	return batch, nil
+}
+
+// spinUpBatchDB spins up a batch that already existed in storage, then
+// returns it.
+func (br *Batcher) spinUpBatchFromDB(ctx context.Context, batch *Batch) error {
+	cfg := BatchConfig{
+		MaxTimeoutDistance: batch.Cfg.MaxTimeoutDistance,
+		BatchConfTarget:    defaultBatchConfTarget,
+	}
+
+	rbfCache := RBFCache{
+		LastHeight: batch.RbfCache.LastHeight,
+		FeeRate:    batch.RbfCache.FeeRate,
+	}
+
+	dbSweeps, err := br.store.FetchBatchSweeps(ctx, batch.ID)
+	if err != nil {
+		return err
+	}
+
+	primarySweep := dbSweeps[0]
+
+	sweeps := make(map[lntypes.Hash]Sweep)
+
+	for _, dbSweep := range dbSweeps {
+		sweep, err := br.convertSweep(dbSweep)
+		if err != nil {
+			return err
+		}
+
+		sweeps[sweep.SwapHash] = *sweep
+	}
+
+	batchKit := BatchKit{
+		ID:               batch.ID,
+		batchTxid:        batch.BatchTxid,
+		batchPkScript:    batch.BatchPkScript,
+		state:            batch.State,
+		primaryID:        primarySweep.SwapHash,
+		sweeps:           sweeps,
+		rbfCache:         rbfCache,
+		returnChan:       br.SweepReqs,
+		completeChan:     br.completeChan,
+		wallet:           br.wallet,
+		chainNotifier:    br.chainNotifier,
+		signerClient:     br.signerClient,
+		musig2SignSweep:  br.musig2ServerSign,
+		verifySchnorrSig: br.VerifySchnorrSig,
+		store:            br.store,
+		log:              batchPrefixLogger(fmt.Sprintf("%d", batch.ID)),
+	}
+
+	newBatch, err := NewBatchFromDB(cfg, batchKit)
+	if err != nil {
+		return err
+	}
+
+	// We add the batch to our map of batches and start it.
+	br.batches[batch.ID] = newBatch
+
+	br.wg.Add(1)
+	go func() {
+		defer br.wg.Done()
+
+		err := newBatch.Run(ctx)
+		if err != nil {
+			br.errChan <- err
+		}
+	}()
+
+	return nil
+}
+
+// FetchUnconfirmedBatches fetches all the batches from the database that are
+// not in a confirmed state.
+func (br *Batcher) FetchUnconfirmedBatches(ctx context.Context) ([]*Batch,
+	error) {
+
+	dbBatches, err := br.store.FetchUnconfirmedSweepBatches(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	batches := make([]*Batch, 0, len(dbBatches))
+	for _, bch := range dbBatches {
+		bch := bch
+
+		batch := Batch{}
+		batch.ID = bch.ID
+
+		switch bch.BatchState {
+		case "open":
```

Comment:
nit: Ideally we'd want to have consts instead of just using these labels everywhere.